### DISCUSSION TITLE
My music files won't fit

### DISCUSSION THREAD

**Discussion by withakw** [rern/rAudio#1272] [Bug Reports]
I have around 150GB of music files when I display them on Windows so I bought a 250gb SD so that I could hold them all on the SD card with rAudio. When I try and copy the files to partition I run out of space.
Does anyone know why this is happening and how I can change the file system so that they will fit

**Comment by ceejayemm** [comment]
rAudio-1 will generally expand the file system to fill the SD card on first boot.  You would be better off getting a smaller SD card (8GB is more than enough) and ONLY use it for the rAudio image. Get a USB flash drive, small SSD or share your music from a NAS or other similar device - use this as your Music Library.

If you put all your music on your (large) SD card if (when?) you need to reflash a new version of rAudio-1 to the SD card you will lose ALL your music files and have to start over with a copy from wherever you hold them now. Using a separate drive/share for your music library saves you all that grief. The expense (?) is worth it!

Chris

**Reply by withakw** [reply]
@ceejayemm I hadn't realised that updating the software would wipe out the music files. Thank you for letting me know

**Comment by R101G** [comment]
@withakw Are you sure this is not a fake SD card? There are a lot of "cheap" cards around which have been doctored to report more space than they actually have.

**Reply by withakw** [reply]
@R101G The packaging says it is a SanDisk card and I bought it from Amazon so I would hope that it is not a fake one. I will check it out. Thank you for responding

**Comment by R101G** [comment]
@withakw Does your Windows write to an ext4 partition? You do not say how much data is transferred or how you are doing it, but I guess you are not inadvertently writing to the boot partition?

**Reply by withakw** [reply]
I was writing 150GB and in the df display it showed for /dev/root that 238GB was used of 238GB. I had a powered USB drive attached with the files on it and I was using cp with the -v option so I could see the files being transferred. There were about 15 albums that were left not transferred

**Reply by R101G** [reply]
That does seem odd. Even allowing for ext4 overhead, I can't see why that should be.

**Comment by withakw** [comment]
Following the recomendation of @ceejayemm I have formatted the SD card as NTFS and I have put the music files on it and put it into an adapter so that it can be added as a USB drive instead but rAudio does not seem to recognise it. I have another 2GB drive formatted as FAT32 which is recognised. Is NTFS supported?

**Reply by R101G** [reply]
I use a NTFS 128GB USB stick with no problem.

**Reply by ceejayemm** [reply]
As rAudio-1 is Linux based I would format your Music Library SD card as EXT4. for best results, however I believe NTFS should work. 

I am now confused, are you using the 2GB SD card for the RAudio-1 image ? If so I think you have taken my suggestion too literally as you will need some working space on this drive. I would go for an 8GB SD card and use a 'proper' USB flash drive to hold your Music Library if that is the way you want to go. 

Once your Music Library is mounted and accessible in rAudio-1 you could use SFTP or the inbuilt File Sharing (Features / Others / File Sharing) to allow your Library to be mapped to a Windows machine to allow new music to be uploaded.

Chris

**Reply by withakw** [reply]
@ceejayemm I have an 8GB card now with the rAudio image and I meant that when it was running I could plug in a 2GB USB stick formatted as FAT32 but when I plug in my 256GB SD card in the SD to USB converter it is not recognised.
I will look at the file sharing options. Thank you

**Comment by R101G** [comment]
@withakw  It is just possible, but very unlikely, that the adapter does not have a driver. That is if another card does not work with the same adapter, of course. Although you transferred the full 150GB to the card formatted to NTFS, that is without the OS so, just as a check, you might try adding 50GB more files to see if it takes them.

**Comment by withakw** [comment]
Hi @R101G and @ceejayemm 
This is the latest state of play. I rebuilt rAudio from scratch on the 256gb sd card and I copied all the 150g files to the SD card using this command
cp -Rn -v /mnt/MPD/USB/PiMusic/'Music FLAC' /mnt/MPD/SD .
This time the space taken on the SD card is 154GB which is a lot better than filling it.

![image](https://user-images.githubusercontent.com/22167274/226360674-8aa5025c-675a-4667-baf6-e784623a5593.png)

If I cd to the SD folder and do an ls it shows nothing

![image](https://user-images.githubusercontent.com/22167274/226362006-8503f68e-761c-4f17-baf8-6ca345a7213f.png)

I have found that my 2 folders of music files are in the root folder

What am I missing? How do I get the files to show up

**Reply by R101G** [reply]
@withakw I can't help feeling there is something wrong with that 256GB card. Can you read it OK in another machine?

EDIT: Presumably you can mount the original USB drive and see the files OK?

**Reply by ceejayemm** [reply]
There is something wrong with your copy command if the files ended up in '/'. Or do you mean the Root partition ?  If you did use the command noted above then I can't see whats wrong but the music files do need to be in  /mnt/MPD/SD for rAudio-1 to see them. 

Sorry out of ideas now.

Chris

**Reply by withakw** [reply]
@R101G
The 256GB card is fine. In the last days I have reformatted it to NTFS and I have loaded all my music files to it so it seems OK on that point. 
@ceejayemm 
When I run the command
cp -Rn -v /mnt/MPD/USB/PiMusic/Music FLAC /mnt/MPD/SD . 
The files are put in /root
I then have to move them to /mnt/MPD/SD and rAudio then picks them up.

I don't understand why the copy command is putting them to /root

Anyway I think I am sorted now

Thank you both for your help and time
